omero/sysadmins/version-requirements.txt

```diff
@@ -0,0 +1,1606 @@
+of enterprise releases, I'd suggest we should only support the latest
+one at any given time or else it ties us into very old dependencies;
+6.x is already very long in the tooth.  I'd suggest moving to 7 once
+it's released and tested by us.  There is extensive CI support for
+building and deployment with CentOS 6 but it's not used directly by
+developers in general.
+
+Linux (Fedora)
+^^^^^^^^^^^^^^
+
+`General overview <https://fedoraproject.org/wiki/Fedora_Release_Life_Cycle>`__
+
+Due to the fast pace of Fedora development, it only really makes sense
+to support the current and perhaps the previous release; these will be
+covered in the dependencies, below.
+
+Linux (Ubuntu)
+^^^^^^^^^^^^^^
+
+`General overview <https://wiki.ubuntu.com/Releases>`__
+
+.. tabularcolumns:: |l|l|l|l|l|J|
+
+.. list-table::
+    :header-rows: 1
+
+    * - Version
+      - Release date
+      - Upstream support
+      - OME (build)
+      - OME (deploy)
+      - Details
+    * - 10.04 (.4) LTS
+      - from Apr 2010
+      - to Apr 2015 (server only)
+      - Supported
+      - Supported
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 10.10
+      - from Oct 2010
+      - to Apr 2012
+      - Dropped
+      - Dropped
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 11.04
+      - from Apr 2011
+      - to Oct 2012
+      - Dropped
+      - Dropped
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 11.10
+      - from Oct 2011
+      - to May 2013
+      - Dropped
+      - Dropped
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 12.04 (.4) LTS
+      - from Apr 2012
+      - to Apr 2017
+      - Supported
+      - Supported
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 12.10
+      - from Oct 2012
+      - to May 2014
+      - Dropped
+      - Dropped
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 13.04
+      - from Apr 2013
+      - to Jan 2014
+      - Dropped
+      - Dropped
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 13.10
+      - from Oct 2013
+      - to Jul 2014
+      - Dropped
+      - Dropped
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 14.04 LTS
+      - from Apr 2014
+      - to Apr 2019
+      - Recommended
+      - Recommended
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 14.10
+      - from Oct 2014
+      - to Jul 2015
+      - Supported
+      - Supported
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+    * - 15.04
+      - TBA
+      - TBA
+      - Upcoming
+      - Upcoming
+      - `Link <https://wiki.ubuntu.com/Releases>`__
+
+Only the LTS and 13.10 releases are supported.  We may wish to only
+formally support e.g. the last one or two LTS releases (being a bit
+more frequent than CentOS/RHEL; 10.04 has quite old dependencies and
+could be considered for dropping.  No CI testing for any version
+except 10.04 on howe/gretzky; some developer use of 12.04 LTS, 14.04
```

Comment:
We could do, but I don't think there's a huge value in doing so.  It's just referring to the general facts that we have CI builds for CentOS (most OMERO and Bio-Formats jobs), and a smaller number of MacOS and Windows jobs, rather than any specific jobs.  The main point is that there is zero testing of the other platforms (Ubuntu, Debian, etc.).

The main thing I'd like to have in place here are regular or semi-regular builds and tests on all the platforms we claim to have support for.  Docker was supposed to help here for the Linux variants, but we're waiting on that at present.  Relating to our policies, we should probably drop Recommended to Supported if we don't do proper testing on that platform.